activesupport/lib/active_support/messages/metadata.rb

```diff
@@ -67,6 +67,14 @@ def match?(purpose)
         def fresh?
           @expires_at.nil? || Time.now.utc < @expires_at
         end
+
+        def parse_expires_at(expires_at)
+          if ActiveSupport.use_standard_json_time_format
+            Time.iso8601(expires_at)
+          else
+            Time.parse(expires_at)
+          end
```

Comment:
You might consider reducing complexity by having your parse method handle whatever value is being assigned.

```ruby
          return expires_at unless expires_at.is_a?(String)
          return Time.iso8601(expires_at) if ActiveSupport.use_standard_json_time_format
          Time.parse(expires_at)
```

Reply:
I thought about it, in most cases `ActiveSupport.use_standard_json_time_format` would be `true`, it's better to short-circuit above, instead of passing it here.

Reply:
@allolex No. The implemented version looks simple to read.